tools/cell_census_builder/validate.py

```diff
@@ -311,6 +318,8 @@ def validate_X_layers(
                 assert X.schema.field("soma_data").type == CENSUS_X_LAYERS[lyr]
                 assert X.shape == (n_obs, n_vars)
 
+                assert _validate_X_has_no_dup_coords(X, n_obs), f"duplicate coordinates in {eb.name} layer {lyr} matrix"
```

Comment:
As it turns out, `_validate_X_layers_contents` catches the dup coords as well (e.g. `AssertionError: homo_sapiens:4 'raw' nnz mismatch 3618682 vs 1809341`, on a test corpus that duplicates every obs row). So this check might not be strictly necessary, though it does make the semantics of the check explicit.